### DISCUSSION TITLE
how about adding QSPI support for such as ST77903 screen

### DISCUSSION THREAD

**Discussion by qiqiji** [Bodmer/TFT_eSPI#2978] [New processor/display support or capability request]
Dear bodmer,

I have been using TFT_eSPI on a few ESP32 projects, and I'm truly impressed by the dedication and quality of your contributions.

Previously, I used the ST7789 display with TFT_eSPI in one of my project, and it worked perfectly. Now, I need to replace the screen with the ST77903, which uses the QSPI interface. However, I noticed that the current TFT_eSPI library does not support it. I found the corresponding driver in the https://github.com/espressif/esp-iot-solution repository and tested it with esp-idf. But without TFT_eSPI, I'm finding it challenging and even not possible to port the UI code from my previous project to the new one. I apologize for taking up your time, but i hope if you have any interest or plans to add QSPI support to TFT_eSPI and that will be great. Thank you.

FYI, driver under esp-idf, i tested works fine ( it cames with lvgl port)
https://github.com/espressif/esp-dev-kits/files/12249442/ST77903_QSPI_1_6_400_400.zip

this one may be newer but i cant compiling
https://github.com/espressif/esp-iot-solution/tree/a3a1caf09bd4026930521bf9a5a9fd529e2e1cbd/components/display/lcd/esp_lcd_st77903

**Comment by libang18** [comment]
I am interested too.

**Reply by CoolmdXi** [reply]
Lots of products, lilly go and many chinese boards now using QSPI. SO would be a bonus if it was added

**Comment by tb205gti** [comment]
I'm also in for the Qspi. The Guition JC3636w518 is a cheap(ish) 1.8" display with SD card, audio in/out build on an ESP32-S3. And assembled in a neat alumim´nium case. Perfekt for esp32 projects.. I really would like tft_espi to support the qspi interface.

**Comment by plain** [comment]
➕. The JC3636W518 / ST77916 setup is an excellent device for prototyping and QSPI support in TFT_eSPI would be great.
